src/components/Schedule/SearchBar.tsx

```diff
@@ -1,26 +1,70 @@
 import React, { useState } from "react";
 import { CloseIcon, SearchIcon } from "@chakra-ui/icons";
 import {
+  Flex,
+  Box,
+  FormLabel,
   IconButton,
   Input,
   InputGroup,
   InputRightElement,
+  Radio,
+  RadioGroup,
+  Stack,
 } from "@chakra-ui/react";
 import { CourseSection } from "../../stores/schedule";
+import { Text } from "devextreme-react/linear-gauge";
+// import { RadioGroup } from "devextreme-react";
 
 interface SearchBarProps {
   getTermData: () => CourseSection[] | undefined;
   setScheduleData: (data: CourseSection[]) => void;
 }
-
+var courseFilter = "all";
+var termFilter = "all";
 export const SearchBar = (props: SearchBarProps) => {
   const { getTermData, setScheduleData } = props;
 
-  const [searchInput, setSearchInput] = useState("");
+  let [searchInput, setSearchInput] = useState("");
   const [filtered, setFiltered] = useState("");
 
-  const filterCourses = (clear?: boolean) => {
-    const termData = getTermData();
+  const toggleCourseTypes = (type: string, group: string) => {
+    if (group === "course") courseFilter = type;
+    else termFilter = type;
+
+    if (courseFilter == "all" && termFilter == "all") {
+      //both are all so give full list
+      setSearchInput("");
+      filterCourses(true);
```

Comment:
Just thinking about best ways to tackle these problems, it might help with the search bar problems if you take out the calls to filterCourses() from the toggle filter logic. This way the logic for searching isn't coupled to the logic for filtering. And then you could just create a new toggleCourses() function to handle the filtering separately.  And then for this you would make a new state variable for specifically toggling as well. Just an idea

Reply:
yeah that makes a lot of sense, and would probably look a lot nicer

Reply:
they do have to work together somehow though right? in order to get the filter and the search to work at the same time?